### ISSUE TITLE
File permissions

### ISSUE THREAD

**Issue by KlutzyBubbles** [maxhoesel-ansible/ansible-collection-pterodactyl#139]
I have been trying to use this to install on a fresh 22.04 image that was downloaded today (and every subsequent run was after i deleted and re created the vm to avoid any other issues), originally it was having permission issues creating /var/www so i just added become: true to the whole role, which seemed to install everything fine but i think when it actually went to launch, the service that this role creates didnt have permission to actually start pterodactyl panel, so i just kept on getting a 500 error when trying to access the domain.

Some questions that i am confused about

Can this role be used with become: true?

What can be done about getting permission denied errors when it tried to create anything in /var/www

If you need anymore information please let me know, i dont exactly know what would be the issue so i dont want to include useless information.

**Comment by KlutzyBubbles** [comment]
Well after trying some more, it does work with become:true but only if i go to the machine and manually run the composer and php artisan commands because for some reason i think the way i have run it or something doesnt allow for creating the vendor folder.

**Comment by maxhoesel** [comment]
Hi, sorry for the late response! Yeah, the roles in this collection are meant to be run with `become: true` or as the root user. That's what the automatic tests do and its what I do for my own panel at home. 

That error does sound odd, what version of the panel were you downloading? The role should have the permissions it needs to install the panel in `/var/www`. If you still have the playbook that you used to install the panel, I'd be happy to take a look

**Comment by KlutzyBubbles** [comment]
> Hi, sorry for the late response! Yeah, the roles in this collection are meant to be run with `become: true` or as the root user. That's what the automatic tests do and its what I do for my own panel at home.
> 
> That error does sound odd, what version of the panel were you downloading? The role should have the permissions it needs to install the panel in `/var/www`. If you still have the playbook that you used to install the panel, I'd be happy to take a look

Not going to lie it has been a while and it has slipped through my mind when i fixed it manually, but looking back at the logs, this is the playbook i was using

```yaml
- hosts: all
  become: true
  tasks:
    - name: Create ssl directory if it does not exist
      ansible.builtin.file:
        path: "/etc/pfsense_ssl/{{ ansible_fqdn }}"
        state: directory
        mode: '0755'
    - name: Create ssh_cert file
      ansible.builtin.copy:
        dest: "/etc/pfsense_ssl/{{ ansible_fqdn }}/cert.pem"
        content: "{{ ssl_cert }}"
    - name: Create ssh_key file
      ansible.builtin.copy:
        dest: "/etc/pfsense_ssl/{{ ansible_fqdn }}/key.pem"
        content: "{{ ssl_key }}"
    - name: Install Pterodactyl Panel
      include_role:
        name: maxhoesel.pterodactyl.pterodactyl_panel
      # pterodactyl_panel supports additional options, see it's README for more details.
      vars:
        pterodactyl_panel_app_key: "{{ app_key }}"
        pterodactyl_panel_hashids_salt: "{{ hash_id_salt }}"
        # The timezone in which the panel should operate
        pterodactyl_panel_timezone: Australia/Melbourne
        # DB settings. Make sure that the database + user already exist and are accessible
        pterodactyl_panel_db_host: "192.168.4.4"
        pterodactyl_panel_db_name: "{{ db_name }}"
        pterodactyl_panel_db_user: "{{ db_username }}"
        pterodactyl_panel_db_password: "{{ db_password }}"
        # Paths to the certificate. This defaults to the Let's Encrypt directory for your hosts domain name
        pterodactyl_panel_ssl_cert: "/etc/pfsense_ssl/{{ ansible_fqdn }}/cert.pem"
        pterodactyl_panel_ssl_key: "/etc/pfsense_ssl/{{ ansible_fqdn }}/key.pem"
        # Mail server settings.
        pterodactyl_panel_mail_host: email-smtp.ap-southeast-2.amazonaws.com
        pterodactyl_panel_mail_user: "{{ input_username }}"
        pterodactyl_panel_mail_password: "{{ input_password }}"
        pterodactyl_panel_mail_encryption: tls
        # By default, the role will create a user with the credentials admin/admin.
        # You can change these values below or in the UI once the panel is up and running
        pterodactyl_panel_admin_mail: "replacement@gmail.com"
        #pterodactyl_panel_admin_user: admin
        #pterodactyl_panel_admin_password: admin
```

Again because i cant remember what exactly i did, i am just reading the logs. It looks like i ran the playbook 2 times (on the same revision), the first resulted in this error at 'Run composer install'

```json
{
  "msg": "Failed to set permissions on the temporary files Ansible needs to create when becoming an unprivileged user (rc: 1, err: chmod: invalid mode: ‘A+user:www-data:rx:allow’\nTry 'chmod --help' for more information.\n}). For information on working around this, see https://docs.ansible.com/ansible-core/2.12/user_guide/become.html#risks-of-becoming-an-unprivileged-user",
  "_ansible_no_log": false
}
```

The second errored on 'Restart schedule service'

```json
{
  "msg": "Unable to start service pterosched.service: Job for pterosched.service failed because the control process exited with error code.\nSee \"systemctl status pterosched.service\" and \"journalctl -xeu pterosched.service\" for details.\n",
  "invocation": {
    "module_args": {
      "name": "pterosched.service",
      "state": "restarted",
      "daemon_reload": false,
      "daemon_reexec": false,
      "scope": "system",
      "no_block": false,
      "enabled": null,
      "force": null,
      "masked": null
    }
  },
  "_ansible_no_log": false,
  "changed": false
}
```

And i am assuming that from the first error, the second run error changed because pterodactyl panel was technically 'installed' so the install step was skipped. 

As far as i can tell i also used the same accounts for running ansible and manual commands. If there is any other information that you think might be useful let me know, i can also always try the install again to see if anything has changed (this information is still from the install i did months ago)

**Comment by maxhoesel** [comment]
Aah, that explains what happened (see [this comment](https://github.com/maxhoesel-ansible/ansible-collection-pterodactyl/issues/151#issuecomment-1741279900) for a brief explanation). I've fixed this in #158, so it shouldn't happen again on new installs, though I'm not sure if upgrading from your current install would work.
